### PR TITLE
python3Packages.dbx: 0.4.1

### PR DESCRIPTION
**Pull request by GuillaumeDesforges**
###### Description of changes

Add `dbx`, a CLI for databricks jobs

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -5972,7 +5976,8 @@ in {
 
   pathos = callPackage ../development/python-modules/pathos { };
 
-  pathpy = callPackage ../development/python-modules/path.py { };
+  # backward compatibility in nixpkgs
+  pathpy = self.path;
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:5980]: That belongs into python-aliases.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -2080,6 +2080,8 @@ in {
 
   dbutils = callPackage ../development/python-modules/dbutils { };
 
+  dbx = callPackage ../development/python-modules/dbx { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:2083]: This should only exist in all-packages if it won't be used as a module.

File: pkgs/development/python-modules/path/default.nix [hunk 3]
```diff
@@ -30,21 +30,13 @@ buildPythonPackage rec {
 
   meta = {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/path/default.nix:31]: meta belongs last.

File: pkgs/development/python-modules/path/default.nix [hunk 4]
```diff
@@ -30,21 +30,13 @@ buildPythonPackage rec {
...
 
   # ignore performance test which may fail when the system is under load
   # test_version fails with 3.8 https://github.com/jaraco/path.py/issues/172
-  disabledTests = [ "TestPerformance" ] ++ lib.optionals isPy38 [ "test_version"];
+  disabledTests = [ "TestPerformance" ];
 
   dontUseSetuptoolsCheck = true;
-
-  patches = [
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/path/default.nix:44]: Please remove the unused fetchpatch from inputs.

File: pkgs/development/python-modules/path/default.nix [hunk 5]
```diff
@@ -12,12 +12,12 @@
...
+  pname = "path";
+  version = "15.1.2";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "9f2169633403aa0423f6ec000e8701dd1819526c62465f5043952f92527fea0f";
+    sha256 = "u2Ka79hoJb8hyLz6D4aRpuWr3D5D1Qtib+aqxbE/YLc=";
   };
 
   checkInputs = [ pytestCheckHook pytest-flake8 glibcLocales packaging ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/path/default.nix:23]: `setuptools-scm` belongs into nativeBuildInputs.

Comment (SuperSandro2000) [inline pkgs/development/python-modules/path/default.nix:23]:
```suggestion
  checkInputs = [ pytestCheckHook glibcLocales packaging ];
```
we shouldn't do lint tests.

File: pkgs/development/python-modules/path/default.nix [hunk 6]
```diff
@@ -12,12 +12,12 @@
 }:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/path/default.nix:12]: `isPy38` is unused.

File: pkgs/development/python-modules/dbx/default.nix [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/databrickslabs/dbx";
+    description = "CLI tool for advanced Databricks jobs management";
+    license = {
+      fullName = "DataBricks eXtensions aka dbx License";
+      url = "https://github.com/databrickslabs/dbx/blob/743b579a4ac44531f764c6e522dbe5a81a7dc0e4/LICENSE";
+      free = false;
+      redistributable = false;
+    };
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dbx/default.nix:55]: Please move this where all other licenses are.

File: pkgs/development/python-modules/dbx/default.nix [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    azure-identity
+    ruamel_yaml
+    emoji
+    cookiecutter
+    retry
+    azure-mgmt-datafactory
+    azure-mgmt-subscription
+  ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dbx/default.nix:45]: Missing `pythonImportsCheck` and the repo contains tests. Why can't we run them? If they are not in the tarball we should fetch from GitHub.

File: pkgs/development/python-modules/dbx/default.nix [hunk 9]
```diff
@@ -0,0 +1,58 @@
+{ buildPythonPackage
+, fetchPypi
+, databricks-cli
+, scipy
+, pathpy
+, tqdm
+, mlflow
+, azure-identity
+, ruamel_yaml
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dbx/default.nix:9]:
```suggestion
, ruamel-yaml
```

**Comment by GuillaumeDesforges** [comment]
Closed in favor of https://github.com/NixOS/nixpkgs/pull/182217/files

**Pull request closed without merging**